specification/cognitiveservices/OpenAI.Inference/models/completions/chat_messages.tsp

```diff
@@ -0,0 +1,202 @@
+  """)
+  low: "low",
+
+  @doc("""
+  Specifies that image evaluation should enable the 'high-res' model that may be more accurate for highly detailed
+  images but may also be slower and consume more tokens.
+  """)
+  high: "high",
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A structured chat content item containing an image reference.")
+model ChatMessageImageContentItem extends ChatMessageContentItem {
+  @doc("The discriminated object type: always 'image_url' for this type.")
+  type: "image_url";
+
+  @doc("An internet location, which must be accessible to the model,from which the image may be retrieved.")
+  url: url;
+
+  @doc("""
+  The evaluation quality setting to use, which controls relative prioritization of speed, token consumption, and
+  accuracy.
+  """)
+  detail?: ChatMessageImageDetailLevel;
+}
+
+@discriminator("role")
+@added(ServiceApiVersions.v2023_05_15)
+@doc("An abstract representation of a chat message as provided in a request.")
+model ChatRequestMessage {
+  @doc("The chat role associated with this message.")
+  role: ChatRole;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("""
+A request chat message containing system instructions that influence how the model will generate a chat completions
+response.
+""")
+model ChatRequestSystemMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'system' for system messages.")
+  role: ChatRole.system;
+
+  @doc("The contents of the system message.")
+  content: string;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A request chat message representing user input to the assistant.")
+model ChatRequestUserMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'user' for user messages.")
+  role: ChatRole.user;
+
+  @doc("The contents of the user message, with available input types varying by selected model.")
+  content: ChatMessageContent;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A request chat message representing response or action from the assistant.")
+model ChatRequestAssistantMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'assistant' for assistant messages.")
+  role: ChatRole.assistant;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly nullable in mirrored API"
+  @doc("The content of the message.")
+  content: string | null;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+
+  @projectedName("json", "tool_calls")
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @doc("""
+  The tool calls that must be resolved and have their outputs appended to subsequent input messages for the chat
+  completions request to resolve as configured.
+  """)
+  toolCalls?: ChatCompletionsToolCall[];
```

Comment:
I think that the yaml uses specifically the concretization `ChatCompletionsFunctionToolCall`. Seems to always be a `chatCompletionMessageToolCall` (yaml file name)

Reply:
IMO the specs are wrong -- Azure matches [OpenAI's spec](https://github.com/openai/openai-openapi/blob/a7a0b63d1fac3acac9a223d57da1219c8a97c01e/openapi.yaml#L5833C15-L5833C15) in this regard, with "function" inside of the generic "tool call," but that's at odds with the pattern where the discriminator-named property only shows up in the hard-coded concrete type that has that name. It's "a nice simplification" for now from the REST structure, but if/when a new tool type is introduced, anticipating the correct structure (where the enum gets more than just "function" and the function property has to move to an extended model) avoids a breaking change in generated object models.